lib/graphql/schema/input_object.rb

```diff
@@ -184,21 +184,25 @@ def validate_non_null_input(input, ctx)
             end
           end
 
-          visible_arguments_map = warden.arguments(self).reduce({}) { |m, f| m[f.name] = f; m}
-
-          # Items in the input that are unexpected
-          input.each do |name, value|
-            if visible_arguments_map[name].nil?
-              result.add_problem("Field is not defined on #{self.graphql_name}", [name])
+          # Inject missing required arguments
+          required_inputs = self.arguments.reduce({}) do |m, (argument_name, argument)|
+            if input[argument_name].nil? && argument.type.non_null? && warden.get_argument(self, argument_name)
```

Comment:
`input[argument_name].nil?` or `!input.key?(argument_name)`

Reply:
Yah this is likely a better way to do it and more explicit on what I am checking. Will update.